### PR TITLE
feat: support cert updates on an APIGW domain name

### PR DESCRIPTION
**Pull request by onebytegone**

### CODE REVIEW

File: src/APIGatewayDomainName.js [hunk 1]
```diff
@@ -28,8 +28,53 @@ module.exports = BaseResource.extend({
          .thenResolve({ PhysicalResourceId: resourceID });
    },
 
-   doUpdate: function(resourceID) {
-      throw new Error(`Updates for custom domains are not supported (${resourceID})`);
+   doUpdate: function(resourceID, props, oldProps) {
+      const cloneOfNewProps = JSON.parse(JSON.stringify(props)),
+            cloneOfOldProps = JSON.parse(JSON.stringify(oldProps));
+
+      // Remove the properties that *can* be updated (there's probally more)
```
Comment (jthomerson) [inline src/APIGatewayDomainName.js:35]:
```suggestion
      // Remove the properties that *can* be updated (there's probably more)
```

File: src/APIGatewayDomainName.js [hunk 2]
```diff
@@ -28,8 +28,53 @@ module.exports = BaseResource.extend({
...
+      delete cloneOfNewProps.regionalCertificateArn;
+      delete cloneOfOldProps.regionalCertificateArn;
+
+      // Then check to see if any of the properties that remain (i.e. those that cannot be
+      // updated) were changed
+      if (!_.isEqual(cloneOfNewProps, cloneOfOldProps)) {
+         // Log the old vs new props rather than putting the data in the error message as
+         // the response can only 4,096 bytes long. See:
```
Comment (jthomerson) [inline src/APIGatewayDomainName.js:43]:
```suggestion
         // the response can only be 4,096 bytes long. See:
```

**Review by jthomerson — CHANGES_REQUESTED** [review]
Couple little typos. There's also the build problem, which I didn't really look at yet.

**Comment by onebytegone** [comment]
The close/reopen was to get travis to rerun the PR build.

**Review by jthomerson — APPROVED** [review]

**Pull request merged**